webapp/channels/src/packages/mattermost-redux/src/utils/data_loader.ts

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/**
+ * A DataLoader is an object that can be used to batch requests for fetching objects from the server for performance
+ * reasons.
+ */
+abstract class DataLoader<Identifier, Result = unknown> {
```

Comment:
I'm not a huge fan of abstract classes in this context. Is there a scenario where we consolidate this into a single class? (Especially if we can merge the interval and delayed data loaders together)?

(A Saga function would be even better, but I agree with using the tools we have for now ;P)

Reply:
As long as those two cases are separate, I think this is fine as-is. I'd rather have two subclasses with their own polling behaviours than to have one class with two different polling behaviours mixed together